compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1132,6 +1315,10 @@ private Integer getTypeHash(BType type) {
         return Objects.hash(type.tag, type.name);
     }
 
+    private boolean getTypeEquality(BType typeA, BType typeB) {
+        return typeA.tag == typeB.tag && Objects.equals(typeA.name, typeB.name);
```

Comment:
Will this logic work for subtypes and referedTypes?

Reply:
Updated to chek for assignability instead in https://github.com/ballerina-platform/ballerina-lang/pull/35130/commits/34c4d787ea2d75f7020af780491c470b7df32f25